packages/manager/src/features/CloudPulse/Utils/CloudPulseWidgetUtils.ts

```diff
@@ -103,3 +140,75 @@ export const getCloudPulseMetricRequest = (
           },
   };
 };
+
+const getLabelName = (
+  labelObject: {
+    flags: FlagSet;
+    label: string;
+    serviceType: string;
+    unit: string;
+  },
+  metric: { [lable: string]: string },
+  resources: CloudPulseResources[]
+): string => {
+  // aggregated metric, where metric keys will be 0
+  if (!Object.keys(metric).length) {
+    // in this case return widget label and unit
+    return labelObject.label + ' (' + labelObject.unit + ')';
+  }
+
+  const results =
+    labelObject.flags?.aclpResourceTypeMap?.filter(
+      (obj: CloudPulseResourceTypeMapFlag) =>
+        obj.serviceType === labelObject.serviceType
+    ) ?? [];
+
+  const flag = results?.length ? results[0] : undefined;
+
+  return getDimensionName(metric, flag, resources);
+};
+
+export const getDimensionName = (
+  metric: { [label: string]: string },
+  flag: CloudPulseResourceTypeMapFlag | undefined,
+  resources: CloudPulseResources[]
+): string => {
+  let labelName = '';
+  Object.keys(metric).forEach((key) => {
+    if (key === flag?.dimensionKey) {
+      const mappedName = mapResourceIdToName(
+        metric[flag.dimensionKey],
+        resources
+      );
+
+      if (mappedName?.length) {
+        labelName +=
+          mapResourceIdToName(metric[flag.dimensionKey], resources) + '_';
+      }
+    } else if (metric[key]) {
+      labelName += metric[key] + '_';
+    }
```

Comment:
Nit: What you have works, but don't we need to return the value? We can make it more readable by using `.entries()` and doing the join outside of the loop for all non-empty strings. You can decide if the boolean is necessary, but I would imagine we want to remove empty strings
```suggestion
 return Object.entries(metric)
    .map(([key, value]) => {
      if (key === flag?.dimensionKey) {
        const mappedName = mapResourceIdToName(value, resources);
        return mappedName || '';
      }
      return value || '';
    })
    .filter(Boolean)
    .join('_');
```

Reply:
Thanks for the suggestions. Yes we have to remove the empty strings.